cmd/aperture-agent/Dockerfile

```diff
@@ -2,6 +2,10 @@
 FROM golang:1.20.4-bullseye AS builder
 
 WORKDIR /src
+
+COPY go.mod go.sum ./
+RUN go mod download
```

Comment:
This will slow down local builds (in which we rely on `--mount-type=cache`, but now any change in go.sum will cause full redownload). I'm not sure how to solve this so it works nicely both in CI and locally though :(

Reply:
This does not work anyway, co closing this PR.

Reply:
Trying a different approach, hope it works https://github.com/fluxninja/aperture/pull/2149